spi/control-plane/policy-spi/src/testFixtures/java/org/eclipse/edc/connector/policy/spi/testfixtures/store/PolicyDefinitionStoreTestBase.java

```diff
@@ -90,6 +90,38 @@ void save_alreadyExists() {
                 .extracting(PolicyDefinition::getCreatedAt).isEqualTo(policy2.getCreatedAt());
     }
 
+
+
+    @Test
+    @DisplayName("Update Policy that does not yet exist")
+    void update_policyDoesNotExist() {
+        var id = getRandomId();
+        var policy = getPolicy(id, "target");
+
+        var updated = getPolicyDefinitionStore().update(id, policy);
+        assertThat(updated).isNull();
```

Comment:
The store should return null in this case. An exception should be raised, or this should be modeled explicitly as a response.

Reply:
That behavior does not seem correct. Let's as @paullatzelsperger why that is the case.

Reply:
similarly to the comment above, issue https://github.com/eclipse-edc/Connector/issues/2531 should take care of that